vault/resource_gcp_auth_backend_role.go

```diff
@@ -252,7 +252,7 @@ func gcpAuthResourceRead(d *schema.ResourceData, meta interface{}) error {
 
 	d.Set("ttl", resp.Data["ttl"])
 	d.Set("max_ttl", resp.Data["max_ttl"])
-	d.Set("type", resp.Data["role_type"])
+	d.Set("type", resp.Data["type"])
```

Comment:
What about handling both forwards and backwards compatibility? So, for each of them, it would basically be, if `resp.Data["whatever"]` has a value, then `d.Set` it? We could just add a comment that it's for compatibility.

Reply:
Good idea. I will try to leverage some time for this. We should probably check for 'role_type' (original value), 'role' (wrong value which has been used in at least one of the 0.11 vault versions and 'type' (correct current value).

Reply:
Thanks!